src/views/Profile/PersonalInfo/RegionCityHandler.tsx

```diff
@@ -0,0 +1,172 @@
+import { Grid, TextField } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import i18n, { langTokens } from 'locales/localizationInit';
+import { RootStateType } from 'models/rootReducer';
+import { IRegion, ICity } from 'old/lib/types';
+import {
+  getCitiesByRegionId,
+  getRegionByCityId,
+} from 'old/lib/utilities/API/api';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { ErrorField } from './ErrorField';
+import { useStyles } from './styles/PersonalInfo.styles';
+import { IRegionCityHandlerProps } from './types';
+
+const RegionCityHandler = ({
+  newAuthorValues,
+  errorFields,
+  visitFields,
+  setNewAuthorValues,
+  setErrorFields,
+  blurHandler,
+}: IRegionCityHandlerProps): JSX.Element => {
+  const classes = useStyles();
+
+  const regions = useSelector(
+    (state: RootStateType) => state.properties.regions,
+  );
+  const cities = useSelector((state: RootStateType) => state.properties.cities);
+  const [selectRegions, setSelectRegions] = useState<IRegion[] | null>(null);
+  const [regionForValue, setRegionForValue] = useState<IRegion | null>(null);
+  const [selectCities, setSelectCities] = useState<ICity[] | null>();
+  const [isCitySelect, setIsCitySelect] = useState<boolean>(false);
+
+  const setRegionByCityId = async (id: number) => {
+    const regionResponse = await getRegionByCityId(id).then((res) => {
+      if (res.status === 200) {
+        return res.data;
+      }
+      return false;
+    });
+    if (regionResponse) {
+      setSelectRegions([regionResponse]);
+      setRegionForValue(regionResponse);
+    }
+  };
+
+  const setCitiesByRegionId = async (id: number) => {
+    const citiesResponse = await getCitiesByRegionId(id).then((res) => {
+      if (res.status === 200) {
```

Comment:
what other status codes possible? I was sure that unlike raw `fetch`, `axios` will not resolve promise for cases like 404 or 500. Or may we here "positive but useless code" like 203?